### PR TITLE
Psychrometrics_Toolkit: Unit Compliance

### LINKED ISSUE #39: Toolkit is not unit compliant
#### Broken rules:

The base SI unit for temperature in the BHoM is Kelvin per: 
https://github.com/BHoM/BHoM/blob/aeb89ef53b6183e8a3865f413e2016404e8a823f/Quantities_oM/Attributes/Temperature.cs#L27-L38

#### Suggestions to restore compliance:

Modify all methods that use temperature as an input or output, modify to the correct unit, K.

### PR DESCRIPTION
**Pull request by albinber**
### Issues addressed by this PR

Closes #39

### CODE REVIEW

**Comment by albinber** [comment]
@bhombot check compliance
@bhombot check call

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
Please use the `typeof(Temperature)` quantity attributes available from `BH.oM.Quantity` [available here](https://github.com/BHoM/BHoM/tree/develop/Quantities_oM/Attributes), rather than putting the units in brackets in the input descriptions 😄

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
Please also add `BHoM/Localisation_Toolkit` as a dependency in the `dependencies.txt` file 😄

File: Psychrometrics_Engine/Compute/DensityWater.cs [hunk 1]
```diff
@@ -29,18 +29,20 @@
...
+using BH.oM.Quantities.Attributes;
 
 namespace BH.Engine.Psychrometrics
 {
     public static partial class Compute
     {
         [Description("Calculates water density from temperature.")]
-        [Input("temperature", "temperature (C).")]
-        [Output("density", "Density (kg/m3).")]
+        [Input("temperature", "Temperature.", typeof(Temperature))]
```
Comment (FraserGreenroyd) [inline Psychrometrics_Engine/Compute/DensityWater.cs:40]: Replace `temperature` with `banana` and...
image

What should the temperature be of? Is there something extra we can provide the user to help them with this input?

File: Psychrometrics_Engine/Compute/FromHumidityRatio.cs [hunk 2]
```diff
@@ -29,21 +29,23 @@
...
     public static partial class Compute
     {
         [Description("Calculates density, enthalpy, dew-point temperature, relative humidity, specific volume and wet-bulb temperature from dry-bulb temperature, pressure and humidity ratio.")]
-        [Input("dryBulbTemperature", "Dry-bulb temperature (C).")]
-        [Input("humidityRatio", "Humidity ratio (kg_water/kg_dryair).")]
-        [Input("pressure", "Air pressure (Pa), defaults to sea level air pressure (101,325 Pa).")]
-        [MultiOutput(0, "density", "Density (kg/m3).")]
-        [MultiOutput(1, "enthalpy", "Enthalpy (J/kg).")]
-        [MultiOutput(2, "dewPoint", "Dew-point temperature (C).")]
+        [Input("dryBulbTemperature", "Dry-bulb temperature.",typeof(Temperature))]
```
Comment (FraserGreenroyd) [inline Psychrometrics_Engine/Compute/FromHumidityRatio.cs:40]:
```suggestion
        [Input("dryBulbTemperature", "Dry-bulb temperature.", typeof(Temperature))]
```

File: Psychrometrics_Engine/Compute/FromHumidityRatio.cs [hunk 3]
```diff
@@ -29,21 +29,23 @@
...
+        [Input("humidityRatio", "Humidity ratio (kg_water/kg_dryair).",typeof(Ratio))]
```
Comment (FraserGreenroyd) [inline Psychrometrics_Engine/Compute/FromHumidityRatio.cs:41]:
```suggestion
        [Input("humidityRatio", "Humidity ratio (kg_water/kg_dryair).", typeof(Ratio))]
```

File: Psychrometrics_Engine/Compute/FromHumidityRatio.cs [hunk 4]
```diff
@@ -29,21 +29,23 @@
...
+        [Input("pressure", "Air pressure, defaults to sea level air pressure (101,325).",typeof(Pressure))]
```
Comment (FraserGreenroyd) [inline Psychrometrics_Engine/Compute/FromHumidityRatio.cs:42]:
```suggestion
        [Input("pressure", "Air pressure, defaults to sea level air pressure (101,325).", typeof(Pressure))]
```

File: Psychrometrics_Engine/Compute/FromHumidityRatio.cs [hunk 5]
```diff
@@ -29,21 +29,23 @@
...
+        [MultiOutput(0, "density", "Density.",typeof(Density))]
```
Comment (FraserGreenroyd) [inline Psychrometrics_Engine/Compute/FromHumidityRatio.cs:43]:
```suggestion
        [MultiOutput(0, "density", "Density.", typeof(Density))]
```

File: Psychrometrics_Engine/Compute/FromHumidityRatio.cs [hunk 6]
```diff
@@ -29,21 +29,23 @@
...
+        [MultiOutput(1, "enthalpy", "Enthalpy.",typeof(SpecificEnergy))]
```
Comment (FraserGreenroyd) [inline Psychrometrics_Engine/Compute/FromHumidityRatio.cs:44]:
```suggestion
        [MultiOutput(1, "enthalpy", "Enthalpy.", typeof(SpecificEnergy))]
```

Comment (FraserGreenroyd) [reply Psychrometrics_Engine/Compute/FromHumidityRatio.cs:44]: Banana again - what is `enthalpy`?

File: Psychrometrics_Engine/Compute/FromHumidityRatio.cs [hunk 7]
```diff
@@ -29,21 +29,23 @@
...
+        [MultiOutput(2, "dewPoint", "Dew-point temperature.",typeof(Temperature))]
```
Comment (FraserGreenroyd) [inline Psychrometrics_Engine/Compute/FromHumidityRatio.cs:45]:
```suggestion
        [MultiOutput(2, "dewPoint", "Dew-point temperature.", typeof(Temperature))]
```

File: Psychrometrics_Engine/Compute/FromHumidityRatio.cs [hunk 8]
```diff
@@ -29,21 +29,23 @@
...
         [MultiOutput(3, "relativeHumidity", "Relative humidity (%).")]
-        [MultiOutput(4, "specificVolume", "Specific Volume (m3/kg).")]
-        [MultiOutput(5, "wetBulbTemperature", "Wet-bulb temperature (C).")]
+        [MultiOutput(4, "specificVolume", "Specific Volume.",typeof(VolumePerQuantity))]
```
Comment (FraserGreenroyd) [inline Psychrometrics_Engine/Compute/FromHumidityRatio.cs:47]:
```suggestion
        [MultiOutput(4, "specificVolume", "Specific Volume.", typeof(VolumePerQuantity))]
```

File: Psychrometrics_Engine/Compute/FromHumidityRatio.cs [hunk 9]
```diff
@@ -29,21 +29,23 @@
...
+        [MultiOutput(5, "wetBulbTemperature", "Wet-bulb temperature.",typeof(Temperature))]
```
Comment (FraserGreenroyd) [inline Psychrometrics_Engine/Compute/FromHumidityRatio.cs:48]:
```suggestion
        [MultiOutput(5, "wetBulbTemperature", "Wet-bulb temperature.", typeof(Temperature))]
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
See initial comments - spaces after the `,` on attribute inputs across the board please - I did the first file then realised it's a pattern so putting back on @albinber and @Tom-Kingstone to resolve please 😄

File: Psychrometrics_Engine/Compute/SaturatedVapourPressureWater.cs [hunk 10]
```diff
@@ -29,26 +29,27 @@
...
+using BH.oM.Quantities.Attributes;
 
 namespace BH.Engine.Psychrometrics
 {
     public static partial class Compute
     {
         [Description("Calculates water SaturatedVapourPressure from temperature.")]
-        [Input("temperature", "temperature (degC).")]
-        [Output("saturatedVapourPressure", "Saturated Vapour Pressure (Pa).")]
+        [Input("temperature", "Temperature.", typeof(Temperature))]
```
Comment (FraserGreenroyd) [inline Psychrometrics_Engine/Compute/SaturatedVapourPressureWater.cs:39]: More bananas?

File: Psychrometrics_Engine/Compute/SpecificHeatCapacityWater.cs [hunk 11]
```diff
@@ -29,21 +29,23 @@
...
 using BH.Engine.Base;
+using BH.oM.Quantities.Attributes;
 
 namespace BH.Engine.Psychrometrics
 {
     public static partial class Compute
     {
         [Description("Calculates Water SpecificHeatCapacity from temperature.")]
-        [Input("temperature", "temperature (C).")]
+        [Input("temperature", "Temperature.", typeof(Temperature))]
```
Comment (FraserGreenroyd) [inline Psychrometrics_Engine/Compute/SpecificHeatCapacityWater.cs:39]: More bananas?

File: Psychrometrics_Engine/Compute/TemperatureAtAltitude.cs [hunk 12]
```diff
@@ -27,18 +27,20 @@
...
+using BH.oM.Quantities.Attributes;
 
 namespace BH.Engine.Psychrometrics
 {
     public static partial class Compute
     {
         [Description("Calculates temperature as a function of altitude.")]
-        [Input("altitude", "altitude (m)")]
-        [Output("temperature", "temperature (C)")]
+        [Input("altitude", "Altitude.", typeof(Length))]
```
Comment (FraserGreenroyd) [inline Psychrometrics_Engine/Compute/TemperatureAtAltitude.cs:38]: More bananas?

File: Psychrometrics_Engine/Compute/TemperatureAtAltitude.cs [hunk 13]
```diff
@@ -27,18 +27,20 @@
...
+        [Output("temperature", "Temperature.", typeof(Temperature))]
```
Comment (FraserGreenroyd) [inline Psychrometrics_Engine/Compute/TemperatureAtAltitude.cs:39]: More bananas?

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

File: Psychrometrics_Engine/Compute/DensityWater.cs [hunk 14]
```diff
@@ -29,21 +29,23 @@
...
+using BH.oM.Quantities.Attributes;
 
 namespace BH.Engine.Psychrometrics
 {
     public static partial class Compute
     {
         [Description("Calculates water density from temperature.")]
-        [Input("temperature", "temperature (C).")]
-        [Output("density", "Density (kg/m3).")]
+        [Input("water temperature", "Water Temperature.", typeof(Temperature))]
```
Comment (FraserGreenroyd) [inline Psychrometrics_Engine/Compute/DensityWater.cs:40]:
```suggestion
        [Input("temperature", "Water Temperature.", typeof(Temperature))]
```

File: Psychrometrics_Engine/Compute/SaturatedVapourPressureWater.cs [hunk 15]
```diff
@@ -29,26 +29,27 @@
...
+using BH.oM.Quantities.Attributes;
 
 namespace BH.Engine.Psychrometrics
 {
     public static partial class Compute
     {
         [Description("Calculates water SaturatedVapourPressure from temperature.")]
-        [Input("temperature", "temperature (degC).")]
-        [Output("saturatedVapourPressure", "Saturated Vapour Pressure (Pa).")]
+        [Input("water temperature", "Water Temperature.", typeof(Temperature))]
```
Comment (FraserGreenroyd) [inline Psychrometrics_Engine/Compute/SaturatedVapourPressureWater.cs:39]:
```suggestion
        [Input("temperature", "Water Temperature.", typeof(Temperature))]
```

File: Psychrometrics_Engine/Compute/SpecificHeatCapacityWater.cs [hunk 16]
```diff
@@ -29,21 +29,23 @@
...
+using BH.oM.Quantities.Attributes;
 
 namespace BH.Engine.Psychrometrics
 {
     public static partial class Compute
     {
-        [Description("Calculates Water SpecificHeatCapacity from temperature.")]
-        [Input("temperature", "temperature (C).")]
+        [Description("Calculates Water SpecificHeatCapacity from water temperature.")]
+        [Input("water temperature", "Water Temperature.", typeof(Temperature))]
```
Comment (FraserGreenroyd) [inline Psychrometrics_Engine/Compute/SpecificHeatCapacityWater.cs:39]:
```suggestion
        [Input("temperature", "Water Temperature.", typeof(Temperature))]
```

File: Psychrometrics_Engine/Compute/TemperatureAtAltitude.cs [hunk 17]
```diff
@@ -27,18 +27,20 @@
...
 
 namespace BH.Engine.Psychrometrics
 {
     public static partial class Compute
     {
-        [Description("Calculates temperature as a function of altitude.")]
-        [Input("altitude", "altitude (m)")]
-        [Output("temperature", "temperature (C)")]
+        [Description("Calculates air temperature as a function of altitude above sea level.")]
+        [Input("altitude above sea level", "Altitude above sea level.", typeof(Length))]
```
Comment (FraserGreenroyd) [inline Psychrometrics_Engine/Compute/TemperatureAtAltitude.cs:38]:
```suggestion
        [Input("altitude", "Altitude above sea level.", typeof(Length))]
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core
@BHoMBot check null-handling

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core
@BHoMBot check serialisation
@BHoMBot check versioning
@BHoMBot check installer

File: dependencies.txt [hunk 18]
```diff
@@ -1,2 +1,4 @@
 BHoM/BHoM
-BHoM/BHoM_Engine
\ No newline at end of file
+BHoM/BHoM_Engine
+BHoM/Units_Engine
+BHoM/Quantities_oM
```
Comment (FraserGreenroyd) [inline dependencies.txt:4]:
```suggestion
BHoM/Localisation_Toolkit
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core

File: dependencies.txt [hunk 19]
```diff
@@ -1,2 +1,4 @@
 BHoM/BHoM
-BHoM/BHoM_Engine
\ No newline at end of file
+BHoM/BHoM_Engine
+BHoM/Units_Engine
```
Comment (FraserGreenroyd) [inline dependencies.txt:3]:
```suggestion
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Review by FraserGreenroyd — APPROVED** [review]
This is looking good now, great work @Tom-Kingstone and @albinber , great bit of pair programming between you both on this 😄

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am requesting neutral checks on: core, installer, null-handling, serialisation, versioning, ready-to-merge

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Pull request merged**